CONTRIBUTING.md

```diff
@@ -0,0 +1,334 @@
+you can use `git commit -a --amend` for that. This will add the current changes to the previous commit.
+Please do not push extra commits that say things like "apply code review" or "fix x" where x is a bug
+introduced in a commit from your PR. In that case always squash the change into the right commit to keep
+the git history clean. Our projects merge the commits as is and will will not squash them on merge to
+preserve the full original context.
+
+### Rebasing
+
+When you created a branch to work on the fix/feature it no longer will be updated with the latest changes
+from the upstream `main` branch. In order to keep your branch up to date you should rebase.
+
+In order to do so add the upstream repo as remote in git, i.e. for containers/common use:
+```
+$ git remote add upstream git@github.com:containers/common.git
+```
+
+Then fetch the latest changes there with
+```
+$ git fetch upstream
+```
+
+And assuming you are still in your fix/feature branch:
+```
+$ git rebase upstream/main
+```
+
+If the PR is open longer you may have to rebase. You must rebase when there is a merge conflict,
+this means the lines that you changed were also changed after you created your branch and in this
+case git does not know what the right change is. You will need to manually resolve it, check
+[here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line)
+for more information on how to do this.
+
+It is recommended to always rebase on a new push to ensure it is testing against the latest code.
+
+## Rust Dependency updates
+
+To automatically keep dependencies up to date we use the [renovate](https://github.com/renovatebot/renovate) bot.
+The bot automatically opens new PRs with updates that should be merged by maintainers.
+
+However sometimes, especially during development, it can be the case that you like to update a dependency.
+
+To do this, you can either run `cargo update` (to update all dependencies) or change the version of the specific dependency you want to update in `Cargo.toml`.
+
+Please run `make` after this to ensure the project still builds after your dependency updates.
+It may be necessary to make code changes to address the updates dependencies.
+
+Then commit the changes and open a PR. If you want to add other changes it is recommended to keep the
+dependency updates in their own commit as this makes reviewing them much easier.
+
+## Find bad changes with git bisect
+
+git bisect is very powerful command in order to quickly find commits that caused a regression.
+
+For example assume you did a Podman update and now something that used to work fine is no longer working,
+this is called a regression. If the change was not intentional it may be hard to find out what caused it.
+git bisect can help with that.
+
+First you need to know the last working version and the the new version were is stopped working and you
```

Comment:
This sentence does not read well.  I think you only need a working version, not the last working version (or you would have your problem nailed down already no?